File: 250226/k번째로 신기한 문자열/kth-special-string.py
n, k, t = input().split()
n, k = int(n), int(k)
str = [input() for _ in range(n)]

strs = list()
for one_str in str:
    include_t = True
    for i in range (len(t)):
        if one_str[i] != t[i]:
            include_t = False
            break
    
    if include_t:
        strs.append(one_str)

strs.sort()
print(strs[k - 1])

File: 250226/k번째로 신기한 문자열/kth-special-string.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    int N, K;
    string T;
    cin >> N >> K >> T;
    
    vector<string> strs;

    for(int i{ 0 };i < N;i++){
        string str;
        cin >> str;
        
        bool include_t{ true };
        for(int j{ 0 };j < T.length();j++){
            if(str[j] != T[j]){ include_t = false; break; }
        }

        if(include_t) strs.push_back(str);
    }

    sort(strs.begin(), strs.end());
    cout << strs[K - 1] << "\n";
}